src/macros.js

```diff
@@ -165,10 +158,22 @@ const getPageVariableMacros = function(string, defaults) {
 
 const replaceMacros = function(string, macros, uriEncode, overrides = {}) {
   for (const macroName in macros) {
+    // The resolvedMacroName is the macro as it appears in the actual string
     const resolvedMacroName = overrides.hasOwnProperty(macroName) ? overrides[macroName] : macroName;
 
-    string = string.split(resolvedMacroName).join(uriEncodeIfNeeded(macros[macroName], uriEncode));
+    if (resolvedMacroName.startsWith('r:')) {
+      try {
+        const regex = new RegExp(resolvedMacroName.slice(2), 'g');
+
+        string = string.replace(regex, uriEncodeIfNeeded(macros[macroName], uriEncode));
+      } catch (error) {
+        videojs.log.warn('Unable to replace regex macro. Please verify that you are passing valid regex.');
```

Comment:
I'm always in favour of making error messages as explicit and useful as possible. What is the effect of being unable to replace the regex macro? What won't work?

Also, is it worth outputting the offending regex?

Reply:
I agree that outputting the input as seen by this function would be helpful to users who may well be programatically generating their regex strings.

Reply:
Thx, it now outputs the offending regex string